remix/app/routes/newsletters.tsx

```diff
@@ -0,0 +1,61 @@
+import type { User } from "@prisma/client";
+import type { LoaderArgs } from "@remix-run/node";
+import {
+  Form,
+  isRouteErrorResponse,
+  useLoaderData,
+  useRouteError,
+} from "@remix-run/react";
+import { SocialsProvider } from "remix-auth-socials";
+import { authenticator } from "~/services/auth.server";
+import type { UserProfile } from "~/types";
+import { db } from "~/utils/db.server";
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const userRequest = (await authenticator.isAuthenticated(request)) as any;
+  const userProfile: UserProfile = userRequest?._json;
+  if (!userProfile) {
+    throw new Response("Forbidden", { status: 403 });
+  }
```

Comment:
Avoid mixing variable definitions and business logic. Any if/else should be placed after variables definitions (unless you work with Go which has a very procedural approach 😉 )

```suggestion
  const userRequest = (await authenticator.isAuthenticated(request)) as any;
  const userProfile: UserProfile = userRequest?._json;
  let user: User | null = null;
  
  if (!userProfile) {
    throw new Response("Forbidden", { status: 403 });
  }
```